### PR TITLE
gic: Move from `setlvl` and `setlvlx` to `intr_enter` and `intr_exit`

### PR DESCRIPTION
**Pull request by richlowe**
These names are at least very likely much more intelligible, and are inspired by the names of the PSM operations which underly them in the intel implementation.  Unlike on intel, I have changed the name throughout the aarch64 code base, rather than kept the old names.

@citrus-it @rmustacc @r1mikey you're probably the ones who might have opinions here?

### CODE REVIEW

**Comment by r1mikey** [comment]
We should definitely talk more :-)

I've been preparing some resource management and GIC-attachment changes in #105 , which lays some of the groundwork for hiding the GIC behind `i86pc` PSM-like interfaces. There's a rambling discussion of where I think this could go in that PR.

TL;DR: I think this is the right direction, but at the same time I think we need to go further and hide the GIC as much as possible.

**Comment by richlowe** [comment]
Right, this is just a low-hanging part of make things better right now.

I have written up what happens now (on other platforms), and what I'd like to begin happening on ARM in https://richlowe.net/interrupts-draft/interrupts.html along with restructuring the code along those lines.

**Comment by richlowe** [comment]
Is anyone against me integrating this as -- I hope -- much more readable names for something that used to be not obvious, even though it doesn't come with a lot of other necessary work?

File: usr/src/uts/armv8/sys/smp_impldefs.h [hunk 1]
```diff
@@ -41,8 +41,27 @@ extern "C" {
...
  */
 extern int (*slvltovect)(int);	/* ipl interrupt priority level */
-extern int setlvl(int);	/* set intr pri represented by vect */
-extern void setlvlx(int);	/* set intr pri to specified level */
+
+/*
+ * Enter the context of the interrupt vector specified as the parameter.
+ *
+ * set interrupt priority to that of the interrupt vector parameter, and
+ * return the old interrupt priority.
```
Comment (rmustacc) [inline usr/src/uts/armv8/sys/smp_impldefs.h:49]: Does this have the same semantics with the  -1 return on spurious interrupt ala x86?

**Review by rmustacc — COMMENTED** [review]
One small question, but I'm fine with the renaming and feel free to land without any follow up from me.

File: usr/src/uts/armv8/sys/smp_impldefs.h (hunk 1 above)
Comment (richlowe) [reply usr/src/uts/armv8/sys/smp_impldefs.h:49]: No, but I can't say for sure it won't ever?

Comment (r1mikey) [reply usr/src/uts/armv8/sys/smp_impldefs.h:49]: This condition should mostly be caught (in a different way) before we hit this code - see the call to `gic_is_spurious` in `do_interrupt` (armv8/os/intr.c). That happens immediately before we call `setlvl` (and that's the only call).
